### PR TITLE
refactor: UpdateManyResults

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
Implement `UpdateManyResults` and use it to implement some `ResultTable` methods.

### CODE REVIEW

File: Adaptors/Memory/src/ResultTable.cs [hunk 1]
```diff
@@ -308,4 +223,47 @@ public Task<HealthCheckResult> Check(HealthCheckTag tag)
...
+                                      ICollection<(Expression<Func<Result, object?>> selector, object? newValue)> updates,
+                                      CancellationToken                                                           cancellationToken = default)
+  {
+    if (!results_.TryGetValue(sessionId,
+                              out var session))
+    {
+      throw new SessionNotFoundException($"Session '{sessionId}' not found");
+    }
+
+    return Task.FromResult(session.AddOrUpdate(resultId,
```
Comment (aneojgurhem) [inline Adaptors/Memory/src/ResultTable.cs:263]: It should be result before update. (It is what's explained in ITaskTable doc). Maybe it returns the value before update ?
If this is based on what's done in TaskTable implementation, there may be an issue and a missing test

File: Adaptors/MongoDB/src/ResultTable.cs [hunk 2]
```diff
@@ -379,6 +273,66 @@ public class ResultTable : IResultTable
                           .ConfigureAwait(false);
   }
 
+  /// <inheritdoc />
+  public async Task<Result> UpdateOneResult(string                                                                      sessionId,
```
Comment (aneojgurhem) [inline Adaptors/MongoDB/src/ResultTable.cs:277]: I don't know if it is time we remove it from the interfaces.
Maybe in another PR

**Review by aneojgurhem — COMMENTED** [review]
Overall, I'm okay with the dev. Thanks !

File: Adaptors/MongoDB/src/ResultTable.cs (hunk 2 above)
Comment (lemaitre-aneo) [reply Adaptors/MongoDB/src/ResultTable.cs:277]: Yes, I think it is better in another PR.

File: Adaptors/Memory/src/ResultTable.cs (hunk 1 above)
Comment (lemaitre-aneo) [reply Adaptors/Memory/src/ResultTable.cs:263]: Yes, it is based on the TaskTable implementation. I will fix both.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**